group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorResult.java

```diff
@@ -66,7 +72,7 @@ public CoordinatorResult(
         List<U> records,
         CompletableFuture<Void> appendFuture
     ) {
-        this(records, null, appendFuture);
+        this(records, null, appendFuture, appendFuture == null);
```

Comment:
I don't like this. I think that we should rather be explicit everywhere about it. How about saying that `replayRecords` is `true` unless specified otherwise?